beacon-chain/casper/validator.go

```diff
@@ -108,3 +108,29 @@ func GetShardAndCommitteesForSlot(shardCommittees []*pb.ShardAndCommitteeArray,
 	}
 	return shardCommittees[slot-lcs], nil
 }
+
+// AreAttesterBitfieldsValid validates that the length of the attester bitfield matches the attester indices
+// defined in the Crystallized State.
+func AreAttesterBitfieldsValid(attestation *pb.AttestationRecord, attesterIndices []uint32) bool {
+	// Validate attester bit field has the correct length.
+	if utils.BitLength(len(attesterIndices)) != len(attestation.AttesterBitfield) {
+		log.Debugf("attestation has incorrect bitfield length. Found %v, expected %v",
+			len(attestation.AttesterBitfield), utils.BitLength(len(attesterIndices)))
+		return false
+	}
+
+	// Valid attestation can not have non-zero trailing bits.
+	lastBit := len(attesterIndices)
+	if lastBit%8 == 0 {
+		return true
+	}
+
+	for i := 0; i < 8-lastBit%8; i++ {
+		if utils.CheckBit(attestation.AttesterBitfield, lastBit+i) {
+			log.Debugf("attestation has non-zero trailing bits")
+			return false
+		}
+	}
+
+	return true
```

Comment:
Why does this return true if a valid attestation must have non-zero trailing bits (line 122)?

Reply:
its part of the spec
>Verify that len(attester_bitfield) == ceil_div8(len(attestation_indices)), where ceil_div8 = (x + 7) // 8. Verify that bits len(attestation_indices).... and higher, if present (i.e. len(attestation_indices) is not a multiple of 8), are all zero